File: client/src/services/common/AuthServices.js
import axios from "axios";

const BASE_URL = import.meta.env.VITE_BACKEND_API;
export const axiosJWT = axios.create();

export const signUp = async (data) => {
  try {
    const res = await axios.post(`${BASE_URL}/auth/register`, data);
    return res.data;
  } catch (error) {
    console.log(error);
  }
};

export const signIn = async (data) => {
  try {
    const res = await axios.post(`${BASE_URL}/auth/login`, data, {
      withCredentials: true,
    });
    return res.data;
  } catch (error) {
    console.log(error);
  }
};

export const refreshToken = async () => {
  try {
    const res = await axios.post(
      `${BASE_URL}/auth/token/refresh`,
      {},
      { withCredentials: true }
    );
    return res.data;
  } catch (error) {
    console.log(error);
  }
};

export const logoutUser = async (accessToken) => {
  try {
    const res = await axios.delete(`${BASE_URL}/auth/logout`, {
      headers: {
        Authorization: `Bearer ${accessToken}`,
      },
      withCredentials: true,
    });
    return res.data;
  } catch (error) {
    console.log(error);
    throw error;
  }
};

File: mobile/lib/core/services/auth_services.dart
import 'package:dio/dio.dart';
import 'package:flutter_dotenv/flutter_dotenv.dart';

class AuthServices {
  final Dio _dio = Dio(
    BaseOptions(
      baseUrl: dotenv.env['BASE_URL'] ?? 'http://localhost:5050/api',
      connectTimeout: Duration(seconds: 10),
      receiveTimeout: Duration(seconds: 10),
      headers: {'Content-Type': 'application/json'},
    ),
  );

  Future<String> login(String email, String password)async {
    try {
      final res = await _dio.post('/auth/login');

      if (res.statusCode == 200) {
        final token = res.data['access_token'];

        return token;
      } else {
        throw Exception(res.data['message'] ?? 'Đăng nhập thất bại');
      }
    } on DioException catch (error) {
      final errorMessage = error.response?.data['message'] ?? error.message;
      throw Exception('Lỗi đăng nhập: $errorMessage');
    }
  }
}